### DISCUSSION TITLE
Compiling for curl x64

### DISCUSSION THREAD

**Discussion by ed-kurlyak** [curl/curl#15277] [General]
I started Power Shell from VS 2019, I run command:

nmake /f Makefile.vc mode=dll debug=yes MACHINE=x64

and compilation started, but then failed with error:

fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'x64'
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX86\x86\link.exe"' : return code '0x458'

Then from Win->Start I run x64 Native Tools Command Prompt VS2019, and get same error:

fatal error LNK1112: тип компьютера модуля "x86" конфликтует с типом целевого компьютера "x64"
NMAKE : fatal error U1077: "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\link.exe" : возвращенный код "0x458"

What is possible todo solve problem?

**Comment by jay** [comment]
> fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'x64'

Most likely you are mistakenly attempting to build 64-bit curl from the x86 native tools prompt. Use "x64 Native Tools Command Prompt for Visual Studio" instead. You will first need to clean the bad build by adding the 'clean' command:

~~~
nmake /f Makefile.vc mode=dll debug=yes MACHINE=x64 clean
nmake /f Makefile.vc mode=dll debug=yes MACHINE=x64
~~~
